### PR TITLE
update reproschema schemas

### PR DESCRIPTION
**Pull request by sanuann**

### CODE REVIEW

**Comment by satra** [comment]
@sanuann - this should be broken up into two schemas: `Field` and `responseOptions` with some of the properties moved to `responseOptions` and the `responseOptions` property of `Field` simply saying it's domain is either a URL or a `responseOptions` object.

for `responseOptions` you will need to further define the property `choices`  as a subtype of `itemListElement`. (and this will require us to update responseOptions on all activities.

File: schemas/responseOptions [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        },
+        {
+            "@id": "reproschema:choices",
+            "@type": "rdf:Property",
+            "rdfs:comment": "An array to list the available options in responseOptions of the Field item.",
+            "rdfs:label": "choices",
+            "schema:domainIncludes": {
+                "@id": "reproschema:responseOptions"
+            },
+            "schema:rangeIncludes": {
```
Comment (sanuann) [inline schemas/responseOptions:153]: @satra what should be the `rangeIncludes` for `choices` ?

File: schemas/Protocol [hunk 2]
```diff
@@ -71,52 +106,158 @@
...
+                {
+                    "@id": "reproschema:Protocol"
+                },
+                {
+                    "@id": "reproschema:Activity"
+                }
+            ],
             "schema:rangeIncludes": {
-                "@id": "reproterms:order"
+                "@id": "schema:Text"
```
Comment (sanuann) [inline schemas/Protocol:162]: @satra what should be range for `addProperties`?

**Comment by sanuann** [comment]
@satra I've updated all the 4 schemas. Please review and let me know if it looks okay.

File: schemas/Protocol (hunk 2 above)
Comment (satra) [reply schemas/Protocol:162]: it should be a new object with those additional properties.

File: schemas/responseOptions (hunk 1 above)
Comment (satra) [reply schemas/responseOptions:153]: range is always about the value of the property - in this case, it should be an object that describes the properties of those choices + skip + don't know.

File: schemas/Activity [hunk 3]
```diff
@@ -51,7 +51,7 @@
             "rdfs:comment": "A description of the item.",
             "rdfs:label": "description",
             "schema:domainIncludes": {
-                "@id": "schema:Thing"
+                "@id": "reproschema:Activity"
```
Comment (satra) [inline schemas/Activity:54]: you should not change any property that you are directly borrowing from schema.org

Comment (satra) [reply schemas/Activity:54]: so replace any schema.org property with their objects.

File: schemas/Activity [hunk 4]
```diff
@@ -191,33 +209,52 @@
             ]
         },
         {
-            "@id": "reproterms:visibility",
+            "@id": "reproschema:variableName",
```
Comment (satra) [inline schemas/Activity:212]: this is not a property of activity.

File: schemas/Field [hunk 5]
```diff
@@ -63,10 +63,10 @@
         {
             "@id": "schema:schemaVersion",
             "@type": "rdf:Property",
-            "rdfs:comment": "Indicates (by URL or string) a particular version of a schema.",
+            "rdfs:comment": "Indicates (by URL or string) a particular version of the reproschema.",
             "rdfs:label": "schemaVersion",
             "schema:domainIncludes": {
-                "@id": ""
+                "@id": "reproschema:Field"
```
Comment (satra) [inline schemas/Field:69]: we should not redefine properties derived from schema.org

File: schemas/Field [hunk 6]
```diff
@@ -147,6 +144,69 @@
...
+                {
+                    "@id": "schema:URL"
+                }
+            ],
+            "schema:sameAs": {
+                "@id": "schema:isPartOf"
+            }
+        },
+        {
+            "@id": "reproterms:responseOptions",
```
Comment (satra) [inline schemas/Field:172]: responseoptions i think should be in reproschema

**Comment by satra** [comment]
`Activity` schema missing `Activity` itself, which should be subclassed from at least CreativeWork in schema.org, unless there is something more specific. we should also add `prov:Entity`, `prov:Plan` and `nidm:Assessment` as a type for an `Activity`.

for `Protocol` we should add the type `prov:Plan`

for `Field` we should add `prov:Entity` and `nidm:DataElement`

File: schemas/Protocol [hunk 7]
```diff
@@ -12,6 +12,29 @@
...
+            "@id": "skos:prefLabel",
+            "@type": "rdf:Property",
+            "rdfs:comment": "The preferred name of the assessment.",
+            "rdfs:label": "prefLabel",
+            "schema:domainIncludes": [
+                {
+                    "@id": "reproschema:Protocol"
+                },
+                {
+                    "@id": "reproschema:Activity"
```
Comment (satra) [inline schemas/Protocol:25]: you should not have to add activity and field here, but it's ok.

**Comment by satra** [comment]
it seems that each schema is actually missing the graph elements for the specific schema itself. here is an example with just that graph element (https://schema.org/actionOption.jsonld)

File: schemas/Activity [hunk 8]
```diff
@@ -151,90 +139,59 @@
...
-            "schema:rangeIncludes": [
-                {
-                    "@id": "reproschema:Activity"
-                },
-                {
-                    "@id": "reproschema:Field"
-                }
-            ]
+            "schema:rangeIncludes": {
+                "@id": "reproschema:addProperties"
```
Comment (sanuann) [inline schemas/Activity:173]: @satra is this correct? both `@id` of this property and `rangeIncludes` point to `reproschema:addProperties` .. I am a lil' confused if this is correct..? or should i name its schema something else and point `rangeIncludes` to this new name?

File: schemas/Field [hunk 9]
```diff
@@ -147,6 +141,69 @@
...
+        {
+            "@id": "reproterms:inputOptions",
+            "@type": "rdf:Property",
+            "rdfs:comment": "An element (object or by URL)to describe the input options of the Field item.",
+            "rdfs:label": "inputOptions",
+            "schema:domainIncludes": {
+                "@id": "reproschema:Field"
+            },
+            "schema:rangeIncludes": [
+                {
```
Comment (sanuann) [inline schemas/Field:197]: @satra can you please check this? I am not sure what this property means (or how it's different from choices). It has been used in mindlogger activities. For reference see here:
https://github.com/ReproNim/reproschema/blob/master/activities/MindLoggerDemo/items/audio-stimulus#L19

There is also a `media` key in the same example linked above. I don't know what it's usage is. I haven't included that here yet. also, I think it hasn't been represented correctly. Shouldn't it be a list of objects?

Comment (satra) [reply schemas/Field:197]: i don't know what inputOptions would do. if we are not using it, let's remove it.

regarding the media key, i'm assuming there are two places:
1. in responseoptions to allow for choices to be an image/icon/sound
2. for a field to be a video, an audio or an image?

so 1 and 2 would require to different places to add media key. where do we add the image to the map task?

File: schemas/Activity (hunk 8 above)
Comment (satra) [reply schemas/Activity:173]: that should not be the case ever. generally some id cannot point to itself as its range. as an example see this: https://schema.org/text.jsonld

in schema.org camelcase is used for properties while classes are capitalized.

File: schemas/Field (hunk 9 above)
Comment (sanuann) [reply schemas/Field:182]: we have used `schema:image` in `responseOptions` see here: https://github.com/ReproNim/reproschema/blob/master/activities/VoiceTask/items/guided_speech#L18

File: schemas/Activity (hunk 8 above)
Comment (sanuann) [reply schemas/Activity:173]: thanks for the example!

File: schemas/Field (hunk 9 above)
Comment (satra) [reply schemas/Field]: for that specific case it doesn't seem appropriate to put it in `responseOptions`. seems more like a property of the `Field`.

perhaps we can use this property (https://schema.org/associatedMedia) at either the `Field` level or the `options` level

Comment (sanuann) [reply schemas/Field]: ok, I'll add it at the `Field` level.

**Comment by satra** [comment]
before we merge this, can you add the validation updates to this PR, so we can make sure all tests pass before we merge the schema?

**Comment by sanuann** [comment]
Merging this. seems to work when testing locally. will fix any issues after merging.

**Pull request merged**